### ISSUE TITLE
skorch

### ISSUE THREAD

**Issue by pfistfl** [mlr-org/mlr3keras#4]
Should really look at skorch.readthedocs.io. I think this has many things we might want!

**Comment by JackyP** [comment]
Hello - skorch is awesome in Python but is anyone able to run torch in reticulate with Rstudio? I recently tried it and ran into https://github.com/rstudio/reticulate/issues/273

I'm interested to write an mlr3 wrapper for fast.ai and its bag of tricks including a tabular learner but did not get very far due to that.

**Comment by pfistfl** [comment]
Hey, nice to see someone looking into this!

I looked at this for a bit earlier this year, (basically also looking into fast.ai tabular learners) and did not get very far as well.
I then started re-writing some of the fast.ai things for `keras` (i.e. the learning rate range test etc.).

I haven't looked too deep into  actually directly calling fast.ai functions (which would be cool).
I guess this would mainly require to convert R's data.tables into some pd.DataFrame that works with fast.ai?

**Comment by pfistfl** [comment]
Just from the top of my head, what I would like to have from fast.ai that is currently not available in mlr3keras:
- [x] cyclical learning rates
- [x] learning rate range test
- [ ] tabular preprocessing stuff (this should go into mlr3pipelines)
- [ ] callbacks for model debugging 
- [ ] fit_one_cycle etc.
- [x] modern optimizers / initializers 

Any other thoughts?
I feel porting architectures between pytorch and keras should not be such a big issue.

If we can talk to fast.ai via reticulate and that all goes smoothly, this would be an optimal outcome I guess :)
